react/features/base/devices/components/DeviceErrorDialog.web.js

```diff
@@ -0,0 +1,211 @@
+            localStorageKey: 'doNotShowErrorAgain',
+            textKey: 'dialog.doNotShowWarningAgain'
+        };
+        const isMicJitsiTrackErrorAndHasName = micError && micError.name
+            && micError instanceof JitsiTrackError;
+        const isCameraJitsiTrackErrorAndHasName
+            = cameraError && cameraError.name
+                && cameraError instanceof JitsiTrackError;
+        let showDoNotShowWarning = false;
+
+        if (micError && cameraError && isMicJitsiTrackErrorAndHasName
+            && isCameraJitsiTrackErrorAndHasName) {
+            showDoNotShowWarning = true;
+        } else if (micError && isMicJitsiTrackErrorAndHasName
+            && !cameraError) {
+            showDoNotShowWarning = true;
+        } else if (cameraError && isCameraJitsiTrackErrorAndHasName
+            && !micError) {
+            showDoNotShowWarning = true;
+        }
+
+        if (micError) {
+            dontShowAgain.localStorageKey += `-mic-${micError.name}`;
+        }
+
+        if (cameraError) {
+            dontShowAgain.localStorageKey += `-camera-${cameraError.name}`;
+        }
+
+        const cameraJitsiTrackErrorMsg
+            = cameraError
+                ? _getCameraErrorMessageKey(cameraError.name)
+                : undefined;
+        const micJitsiTrackErrorMsg
+            = micError
+                ? _getMicErrorMessageKey(micError.name)
+                : undefined;
+
+        const cameraErrorMsg
+            = cameraError
+                ? cameraJitsiTrackErrorMsg
+                    || _getCameraErrorMessageKey(JitsiTrackErrors.GENERAL)
+                : '';
+        const micErrorMsg
+            = micError
+                ? micJitsiTrackErrorMsg
+                    || _getMicErrorMessageKey(JitsiTrackErrors.GENERAL)
+                : '';
+        const additionalCameraErrorMsg
+            = !cameraJitsiTrackErrorMsg
+                && cameraError
+                && cameraError.message
+                    ? <div>${cameraError.message}</div>
+                    : '';
+        const additionalMicErrorMsg
+            = !micJitsiTrackErrorMsg
+                && micError
+                && micError.message
+                    ? <div>${micError.message}</div>
+                    : '';
+
+        let cameraMessage, micMessage;
+
+        /* eslint-disable react/jsx-wrap-multilines */
+        if (micError) {
+            micMessage
+                = <div>
+                    <h3>{ t('dialog.micErrorPresent') }</h3>
+                    <h4>{ t(micErrorMsg) }</h4>
+                    {additionalMicErrorMsg}
+                </div>;
+        }
+
+        if (cameraError) {
+            cameraMessage
+                = <div>
+                    <h3>{ t('dialog.cameraErrorPresent') }</h3>
+                    <h4>{ t(cameraErrorMsg) }</h4>
+                    {additionalCameraErrorMsg}
+                </div>;
+        }
+        /* eslint-enable react/jsx-wrap-multilines */
```

Comment:
I wonder if it's better to do camera error or mic error first and then do the other error, instead of alternating. What do you think?